src/docs/src/best-practices/jsdevel.rst

```diff
@@ -46,3 +46,254 @@ tips and tricks that will ease the difficulty.
+   typein:1:11  var xml = <root><x></x></root>
+   typein:1:11 ...........^
+
+3. ``toLocaleFormat(...)`` function.
+
+   This ``Date`` function is not present in versions greater than ``1.8.5``:
+
+.. code-block:: bash
+
+   % js
+   js> d = new Date("Dec 1, 2015 3:22:46 PM")
+   (new Date(1449001366000))
+   js> d.toLocaleFormat("%Y-%m-%d")
+   "2015-12-01"
+
+   % js91
+   js> d = new Date("Dec 1, 2015 3:22:46 PM")
+   (new Date(1449001366000))
+   js> d.toLocaleFormat("%Y-%m-%d")
+   typein:2:3 TypeError: d.toLocaleFormat is not a function
+
+4. Invalid expressions following ``function(){...}`` are not ignored any longer
+   and will throw an error.
+
+   Previously, in versions less than or equal to ``1.8.5`` it was possible add
+   any expression following the main function definition and they were mostly
+   ignored:
+
+.. code-block:: bash
+
+   $ http put $DB/db/_design/d4 views:='{"v1":{"map":"function(doc){emit(1,2);} if(x) a"}}'
+   HTTP/1.1 201 Created
+   {
+       "id": "_design/d4",
+       "ok": true,
+       "rev": "1-08a7d8b139e52f5f3df5bc27e20eeff1"
+   }
+
+   % http $DB/db/_design/d4/_view/v1
+   HTTP/1.1 200 OK
+   {
+       "offset": 0,
+       "rows": [
+           {
+               "id": "doc1",
+               "key": 1,
+               "value": 2
+           }
+       ],
+       "total_rows": 1
+   }
+
+  With higher versions of spidermonkey, that would throw a compilation error:
+
+.. code-block:: bash
+
+    $ http put $DB/db/_design/d4 views:='{"v1":{"map":"function(doc){emit(1,2);} if(x) a"}}'
+    HTTP/1.1 400 Bad Request
+    {
+        "error": "compilation_error",
+        "reason": "Compilation of the map function in the 'v1' view failed: ..."
+    }
+
+5. Object key order.
+
+   Object key order may change between versions, so any views which rely on
+   that order may emit different results depending on the engine version:
+
+.. code-block:: bash
+
+   % js
+   js> r={}; ["Xyz", "abc", 1].forEach(function(v) {r[v]=v;}); Object.keys(r)
+   ["Xyz", "abc", "1"]
+
+   % js91
+   js> r={}; ["Xyz", "abc", 1].forEach(function(v) {r[v]=v;}); Object.keys(r)
+   ["1", "Xyz", "abc"]
+
+Using QuickJS
+=============
+
+The QuickJS-based JavaScript engine is available as of CouchDB version 3.4. It
+has to be explicitly enabled via ``[couchdb] js_engine = quickjs`` and
+restarting the service.
+
+Generally, QuickJS engine is a bit faster, consumes less memory, and provides
+slightly better isolation between contexts by re-creating the whole javascript
+engine runtime on every ``reset`` command.
+
+To try building invidual views using QuickJS, even when the default engine is
+SpiderMonkey, can use the ``"javascript_quickjs"`` as the view language,
+instead of ``"javascript"``. Just that view will be rebuilt using the QuickJS
+engine. However, when switching back to ``"javascript"`` the view will have to
+be re-built again.
+
+QuickJS vs Spidermonkey incompatibilities
```

Comment:
"SpiderMonkey"

Reply:
Yup, definitely. Thanks for pointing it out.